### ISSUE TITLE
Fixing Sega 32x

### ISSUE THREAD

**Issue by gr33k** [linuxserver/emulatorjs#117]
I see 32x is broken...What can we do to fix this?

I see libretro has a Web assembly version (i.e. https://web.libretro.com/) can we grab this or another core perhaps?

EmulatorJS has Picodrive but they've changed their code a bit since 4.x so it does not appear to be a drop-in/rename replacement. However, I realized the emulator cores in the .data extension are simply 7zipped and you can extract them to obtain the .js and .wasm files (this is how I renamed them and tried to drop them in to the container, but it didn't work). I tried using 32x.js/wasm and picodrive_libretro.js/wasm

It tries to load when I use the libretro naming convention and config file reference (libretro-picodrive), BUT some issues when inspecting it:
[ERROR] Couldn't find any audio driver named "rwebaudio" 
[ERROR] Couldn't find any input driver named "rwebinput"
(I mention these because these same drivers ARE found in the working cores when checking their logs)
(attached log)
[libretro_picodrive.log](https://github.com/linuxserver/emulatorjs/files/13855437/libretro_picodrive.log)

Attached this in case it's handy
[picodrive-libretro.zip](https://github.com/linuxserver/emulatorjs/files/13855426/picodrive-libretro.zip)

I lastly tried to  compile picodrive itself using emscripten with the same method to test, but that also failed.

Any thoughts? Would love to figure this out and perhaps apply this same method to other libretro cores! Ty

**Comment by gr33k** [comment]
Forget it - Even their core isn't working right now...https://github.com/EmulatorJS/EmulatorJS/issues/579
Hoping someone can get it working from the libretro cores perhaps...sux

**Comment by thelamer** [comment]
Picodrive has always black screened in the emscripten libretro core. I had to EOL the old decompilation stuff so sorry no 32x until we have a viable picodrive core we can build from source.

**Comment by zmarteline** [comment]
I currently use version 3.1.5 for 32X games. It works perfectly, but without the improvements achieved in the newer versions. As a workaround, couldn't you use the v3.1.5 core in emulatorjs 4.x for 32X games?

**Comment by thelamer** [comment]
No, I promised another project I would not ship those old decompilation attempt bins anymore as it misrepresents the current state of their project. You are absolutely free to rip them from here https://github.com/thelamer/emulatorjs/tree/d42600c8f0afad6c5787f12dc188491ffebdd4e4 and place them in your /emulatorjs/frontend directory properly and use them though.

**Comment by gr33k** [comment]
@thelamer I see the .data file but I can't seem to unzip it using 7zip. Is there a way you can provide the .js and .wasm file for the old 32x so we can easily drop it in please?

I can understand you want to get a clean emulator in here vs this obfuscated thing...but a working emulator is better than no working emulator. There isn't one coming out soon either it seems...so I think (if noted in the docs) people would give you a pass for using this until a clean/better option is available. I certainly wouldn't mind ;) ty

**Comment by gr33k** [comment]
Any ETA on when this will be resolved or an easy way to drop-in the old core temporarily?

Thank you!

**Comment by gr33k** [comment]
Any Update on when this will be fixed or easy drop-in replacement of old cores? Thank you!

**Comment by gr33k** [comment]
@thelamer Happy New Year!

It seems that the emulatorJS team has a working 32x core now. I tested a few roms and some DO have audio problems (try Mortal Kombat II for example)...but others seem to work fine so far!  I tested the stable 4.2.1.

Do you think this core can be ported?
(https://demo.emulatorjs.org/)

**Comment by thelamer** [comment]
Yeah for sure, I am wrapped up with work for a bit, but I want to give some cycles in the near future to check the current state of the emus

**Comment by gr33k** [comment]
Great news! I also see a PSP and Amiga core on their list that would be pretty neat 🍺 

Looking forward to it man - Thank you for your work 😄
